### PR TITLE
Allow SearchIndex constructor to init null fields

### LINKED ISSUE #15590: SearchIndexClient.GetIndexNames throws
**Describe the bug**
Calling `SearchIndexClient.GetIndexNames` throws an `ArgumentNullException` with the message "Fields cannot be null. To clear values, call Clear. (Parameter 'value')".

**Expected behavior**
The method should return the list of index names.

**Actual behavior (include Exception or Stack Trace)**
Calling `SearchIndexClient.GetIndexNames` throws an `ArgumentNullException` with the message "Fields cannot be null. To clear values, call Clear. (Parameter 'value')".

The reason this is happening is that `GetIndexNames` calls `IndexesClient.List` with `Constants.NameKey`, which results in a JSON document missing all properties except for the name of the index. Attempting to deserialize this then fails because `fields` are required, and are missing (despite definitely existing on the actual indexes) in the JSON getting passed in.

The implementation of `GetIndexNames` and `GetIndexNamesAsync` should either use a different new method on the `IndexerRestClient` that doesn't attempt to deserialize indexes but just extracts the name that are needed, or should Constants.All before it does the deserialization like `GetIndexes` does.

**Workaround**
Use `SearchIndexClient.GetIndexes(serviceName, credentials).Select(idx => idx.Name)` instead.

**To Reproduce**
Call `SearchIndexClient.GetIndexNames`.

**Environment:**
 - Name and version of the Library package used: Azure.Search.Documents 11.1.1
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): .NET Core 3.1
 - IDE and version : Visual Studio 16.8.0 Preview 3.2

### PR DESCRIPTION
**Pull request by heaths**
Fixes #15590

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs [hunk 1]
```diff
@@ -286,6 +286,22 @@ public async Task GetIndexesNextPageThrows()
...
 
+        [Test]
+        public async Task GetIndexNames()
+        {
+            await using SearchResources resources = await SearchResources.GetSharedHotelsIndexAsync(this);
+
+            SearchIndexClient client = resources.GetIndexClient();
+
+            bool found = false;
+            await foreach (string name in client.GetIndexNamesAsync())
```
Comment (bleroy) [inline sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs:297]: You could simplify this with a single `client.GetIndexNamesAsync().Any(name => string.Equals(resources.IndexName, name, StringComparison.InvariantCultureIgnoreCase))`.

**Review by bleroy — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs (hunk 1 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs:297]: That would be ideal, but it returns an `IAsyncEnumerable`, which LINQ methods don't seem to support.

File: sdk/search/Azure.Search.Documents/src/Generated/Models/SearchIndex.cs [hunk 2]
```diff
@@ -30,10 +30,10 @@ public partial class SearchIndex
...
         /// <param name="Etag"> The ETag of the index. </param>
-        internal SearchIndex(string name, IList<SearchField> fields, IList<ScoringProfile> scoringProfiles, string defaultScoringProfile, CorsOptions corsOptions, IList<SearchSuggester> suggesters, IList<LexicalAnalyzer> analyzers, IList<LexicalTokenizer> tokenizers, IList<TokenFilter> tokenFilters, IList<CharFilter> charFilters, SearchResourceEncryptionKey encryptionKey, SimilarityAlgorithm similarity, string Etag)
+        internal SearchIndex(string name, IList<SearchField> Fields, IList<ScoringProfile> scoringProfiles, string defaultScoringProfile, CorsOptions corsOptions, IList<SearchSuggester> suggesters, IList<LexicalAnalyzer> analyzers, IList<LexicalTokenizer> tokenizers, IList<TokenFilter> tokenFilters, IList<CharFilter> charFilters, SearchResourceEncryptionKey encryptionKey, SimilarityAlgorithm similarity, string Etag)
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Generated/Models/SearchIndex.cs:33]: @pakrym, it's not an issue, but I found this unusual with the generator. When I changed to using a field, the parameter name casing was changed to PascalCase. Still internal/private accessibility, so not a problem.

Comment (pakrym) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/SearchIndex.cs:33]: It's a bug in our identifier generation logic. Can you file it, please?

Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/SearchIndex.cs:33]: https://github.com/Azure/autorest.csharp/issues/925

**Pull request merged**